### DISCUSSION TITLE
Are you going to continue using css-in-js? It negatively affects the performance.

### DISCUSSION THREAD

**Discussion by ghost** [ant-design/ant-design#43668] [General]
The lib is the most capable of UI libs for React I've seen, but after v5 update it became too slow.

How am I supposed to build enterprise-level apps using a tool that has a 3 of 100 performance score on its own documentation page?
https://i.postimg.cc/GtjZxHLb/2023-07-20-01-50-09-ant-design-performance.png

To compare, v4 docs page has 54 performance score
https://i.postimg.cc/QCVp0zsV/2023-07-20-01-54-45-4x-ant-design-performance.png

Can anyone explain me why this happens? I have only one idea: css-in-js is known to be slow and that was the major change in v5.

**Comment by zombieJ** [comment]
Thanks for feedback. The site introduces many unnecessary libraries for testing framework compatibility. We are working to clean this deps up step by step. Sorry for bad experience.

**Comment by abenhamdine** [comment]
FWIW I just a made a test today (in Chrome Incognito mode) after https://github.com/ant-design/ant-design/pull/43705 has landed

Results vary a lot but the difference between v4 and v5 is still huge.
But I don't think the comparison is fair because there are many more examples in the v5 page  

#### antd4 :  https://4x.ant.design/components/form/

5 measures : 61, 84, 38, 76, 68

#### antd5 :  https://ant.design/components/form

5 measures : 15, 17, 18, 27, 23
